contracts/test/psm/PegStabilityModuleNonCustodial.t.sol

```diff
@@ -0,0 +1,235 @@
+pragma solidity ^0.8.4;
+
+import {ERC20, IERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+
+import {MockPCVDepositV2} from "../../mock/MockPCVDepositV2.sol";
+import {IPCVDeposit} from "../../pcv/IPCVDeposit.sol";
+import {MockERC20} from "../../mock/MockERC20.sol";
+import {MockOracle} from "../../mock/MockOracle.sol";
+import {ICore} from "../../core/ICore.sol";
+import {Core} from "../../core/Core.sol";
+import {IFei, Fei} from "../../fei/Fei.sol";
+import {PegStabilityModuleNonCustodial} from "./../../peg/PegStabilityModuleNonCustodial.sol";
+import {PegStabilityModule} from "./../../peg/PegStabilityModule.sol";
+import {Vm} from "./../utils/Vm.sol";
+import {DSTest} from "./../utils/DSTest.sol";
+import {getCore, getAddresses, FeiTestAddresses} from "./../utils/Fixtures.sol";
+
+contract PegStabilityModuleNonCustodialTest is DSTest {
+
+    PegStabilityModuleNonCustodial private psm;
+    ICore private core;
+    IFei private fei;
+
+    modifier prank(address toPrank) {
+      vm.startPrank(toPrank);
+
+      _;
+
+      vm.stopPrank();
+    }
+
+    uint256 immutable public mintAmount = 10_000_000e18;
+    uint256 immutable public bufferCap = 10_000_000e18;
+    uint256 immutable public rps = 10_000e18;
+
+    uint256 immutable public floorBasisPoints = 9_900;
+    uint256 immutable public ceilingBasisPoints = 10_100;
+
+    MockERC20 public underlyingToken;
+    MockPCVDepositV2 public pcvDeposit;
+    MockOracle public oracle;
+
+    Vm public constant vm = Vm(HEVM_ADDRESS);
+    FeiTestAddresses public addresses = getAddresses();
+    address immutable public userAddress = addresses.userAddress;
+
+    function setUp() public {
+      core = getCore();
+
+      fei = core.fei();
+
+      oracle = new MockOracle(1);
+      underlyingToken = new MockERC20();
+      pcvDeposit = new MockPCVDepositV2(
+          address(core),
+          address(underlyingToken),
+          0,
+          0
+      );
+
+      PegStabilityModule.OracleParams memory oracleParams = PegStabilityModule.OracleParams({
+          coreAddress: address(core),
+          oracleAddress: address(oracle),
+          backupOracle: address(0),
+          decimalsNormalizer: 0,
+          doInvert: false
+      });
+
+      psm = new PegStabilityModuleNonCustodial(
```

Comment:
There's quite a bit of setup going on here. I'd maybe add some comments to make it easier to follow or break some of it out into a fixture or other function. For example, the granting of permissions or minting could be some form of `setupPSM()` fixture perhaps

Reply:
added comments documenting this setup.